sanic_jwt/configuration.py

```diff
@@ -1,119 +1,313 @@
-    'refresh_token_name': 'refresh_token',
-    'path_to_authenticate': '/',
-    'path_to_retrieve_user': '/me',
-    'path_to_verify': '/verify',
-    'path_to_refresh': '/refresh',
-    'private_key': None,
-    'scopes_enabled': False,
-    'scopes_name': 'scopes',
-    'secret': 'This is a big secret. Shhhhh',
-    'strict_slashes': False,
-    'url_prefix': '/auth',
-    'user_id': 'user_id',
-    'verify_exp': True,
+    "access_token_name": "access_token",
+    "algorithm": "HS256",
+    "auth_mode": True,
+    "authorization_header": "authorization",
+    "authorization_header_prefix": "Bearer",
+    "authorization_header_refresh_prefix": "Refresh",
+    "claim_aud": None,
+    "claim_iat": False,
+    "claim_iss": None,
+    "claim_nbf": False,
+    "claim_nbf_delta": 0,
+    "cookie_domain": "",
+    "cookie_httponly": True,
+    "cookie_refresh_token_name": "refresh_token",
+    "cookie_set": False,
+    "cookie_strict": True,
+    "cookie_access_token_name": "access_token",
+    "debug": False,
+    "expiration_delta": 60 * 5 * 6,
+    "generate_refresh_token": utils.generate_token,
+    "leeway": 60 * 3,
+    "refresh_token_enabled": False,
+    "refresh_token_name": "refresh_token",
+    "path_to_authenticate": "/",
+    "path_to_retrieve_user": "/me",
+    "path_to_verify": "/verify",
+    "path_to_refresh": "/refresh",
+    "private_key": None,
+    "scopes_enabled": False,
+    "scopes_name": "scopes",
+    "secret": "This is a big secret. Shhhhh",
+    "strict_slashes": False,
+    "url_prefix": "/auth",
+    "user_id": "user_id",
+    "verify_exp": True
 }
 
 aliases = {
-    'cookie_token_name': 'cookie_access_token_name',
-    'public_key': 'secret',
+    "cookie_access_token_name": "cookie_token_name", "secret": "public_key"
 }
 
+ignore_keys = (
+    "add_scopes_to_payload",
+    "authenticate",
+    "class_views",
+    "extend_payload",
+    "retrieve_refresh_token",
+    "retrieve_user",
+    "store_refresh_token",
+)
+
 logger = logging.getLogger(__name__)
-config = None
+
+
+def _warn_key(key):
+    if key not in ignore_keys:
+        logger.warning(
+            "Configuration key '%s' found is not valid for sanic-jwt", key
+        )
+
+
+def _create_or_overwrite_config_item(value, key, item_aliases, instance):
+    setattr(
+        instance,
+        key,
+        ConfigItem(
+            value, item_name=key, config=instance, aliases=item_aliases
+        ),
+    )
+
+
+def _update_config_item(key, item_aliases, instance):
+    getattr(instance, key)._item_name = key
+    getattr(instance, key)._aliases = item_aliases
+    getattr(instance, key)._config = instance
+
+
+class ConfigItem:
+
+    def __init__(
+        self,
+        value,
+        item_name=None,
+        config=None,
+        inject_request=True,
+        aliases=None,
+    ):
+        self._value = value
+        self._item_name = item_name
+        self._config = config
+        self._inject_request = inject_request
```

Comment:
Also `inject_user` as a convenience? It may save on a call to DB.

Reply:
I don't know. The only thing we have right away is `request`. To have `inject_user`, we should first extract the user from the payload, then put it inside the context _which may change_ regarding what  other config keys may look like ... Simple example:

```python
@app.route("/something")
@sanicjwt.protected(authorization_header="foobar")
async def mymethod(request):
    ...
```

This would be way too tricky ...

Reply:
Okay. I'll try an make a utility that hooks into `retrieve_user` but only requires the `request` argument. `extract_user_from_request()` as a convenience.